clients/src/main/java/org/apache/kafka/common/memory/GarbageCollectedMemoryPool.java

```diff
@@ -37,13 +37,13 @@ public class GarbageCollectedMemoryPool extends SimpleMemoryPool implements Auto
     //serves 2 purposes - 1st it maintains the ref objects reachable (which is a requirement for them
     //to ever be enqueued), 2nd keeps some (small) metadata for every buffer allocated
     private final Map<BufferReference, BufferMetadata> buffersInFlight = new ConcurrentHashMap<>();
-    private final GarbageCollectionListener gcListener = new GarbageCollectionListener();
     private final Thread gcListenerThread;
-    private volatile boolean alive = true;
+    private volatile boolean alive;
 
     public GarbageCollectedMemoryPool(long sizeBytes, int maxSingleAllocationSize, boolean strict, Sensor oomPeriodSensor) {
         super(sizeBytes, maxSingleAllocationSize, strict, oomPeriodSensor);
         this.alive = true;
```

Comment:
it seems to me removing `this.alive = true;` is more suitable.

Reply:
Yes I agree